wagon.py

```diff
@@ -26,6 +25,7 @@
 import logging
 import platform
 import tempfile
+import StringIO
```

Comment:
one thing to note though is that io.StringIO mandates unicode, so the code should be aware of that